andela/ah-lobos-backend
# #166789873 Enable social login authentication

#### What does this PR do?
This pull request adds resources that allow users to: 
 - login using their  Facebook accounts
- login using their  Google accounts
- login using their  Twitter accounts

#### Description of Task to be completed?
Enable social login via google, twitter, and Facebook in the application

#### How should this be manually tested?
Follow instructions on the `README.md` file attached to this repo to set up the project on your local machine.
Make sure that you have added all environment variables using `example.env`
- Run the application
```
> npm run dev
```

Access the following routes to test
> Use this routes `/facebook`, `/twitter` and `/google` in the browser
after accessing those route you will be redirected to google, facebook or twitter
 and then login using your social account

Test using Heroku links (click to the link to test):

[Google Login](https://ah-lobos-backend-swagger.herokuapp.com/google)

[Facebook Login](https://ah-lobos-backend-swagger.herokuapp.com/facebook)

[Twitter Login](https://ah-lobos-backend-swagger.herokuapp.com/twitter)

#### Any background context you want to provide?
- Create .env file with the help of example.env
- navigate to the above route to access any of social login you want
#### What are the relevant pivotal tracker stories?
[#166789873](https://www.pivotaltracker.com/story/show/166789873)
#### Screenshots (if appropriate)
- The following are screenshots of login using google, twitter and facebook
![image](https://user-images.githubusercontent.com/31614647/61025428-fd415680-a3b0-11e9-8470-8cf7d999c26e.png)
![image](https://user-images.githubusercontent.com/31614647/61025381-d7b44d00-a3b0-11e9-929e-fb759475a18f.png)
![image](https://user-images.githubusercontent.com/31614647/61025463-1a762500-a3b1-11e9-857a-fd17a9f1de4c.png)
![image](https://user-images.githubusercontent.com/31614647/61025505-485b6980-a3b1-11e9-93b1-431929379401.png)

- mockups:

![image](https://user-images.githubusercontent.com/31614647/61175949-4adbe000-a5b8-11e9-972d-525e29e6cc07.png)

![image](https://user-images.githubusercontent.com/31614647/61175952-516a5780-a5b8-11e9-8295-1b39a3a5ced1.png)


#### Questions:
- N/A
